core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -64,6 +64,19 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
   @volatile private var dirtiestLogCleanableRatio = 0.0
   newGauge("max-dirty-percent", new Gauge[Int] { def value = (100 * dirtiestLogCleanableRatio).toInt })
 
+  /* a gauge for tracking the number of logCleaner runs */
+  @volatile private var numLogCleanRuns = 0
+  newGauge("num-runs", new Gauge[Int] { def value = numLogCleanRuns })
+
+  /* a gauge for tracking the last time of logCleaner run */
+  @volatile private var logCleanLastRun : Long = System.currentTimeMillis
+  newGauge("last-run-time", new Gauge[Long] { def value = logCleanLastRun })
+
+  /* a gauge for tracking the number of filthy logs. A non zero value indicates
```

Comment:
@tcrayford  @wushujames  `time-since-last-run-ms` metric is used to track if the log cleaner thread is alive. `num-filthy-logs` serves slightly different purpose.  It is used to track a scenario where log cleaner tries to clean a log segment but fails to do so, because of an exception. Segment remains in unclean state. This metric is used to track such cases, it doesn't track if the log cleaner threads are keeping up. In an ideal scenario, `num-filthy-logs` should return to 0 in a reasonable amount of time. If it doesn't, then it is time to examine the logs for any exceptions.